### PR TITLE
Refactor mb

### PR DESCRIPTION
**Pull request by Maxwell-Baird**
Fixed routing errors which prevent updating admin/items
Added a test to order_spec testing if a merchant will get the expect return value as an user

### CODE REVIEW

File: app/controllers/merchant/items_controller.rb [hunk 1]
```diff
@@ -26,7 +26,7 @@ def update
     end
   end
 
-  def add_item
+  def create
```
Comment (colintalex) [inline app/controllers/merchant/items_controller.rb:29]: Nice, much more RESTful

File: spec/features/admin/merchants_spec.rb [hunk 2]
```diff
@@ -204,11 +204,11 @@
       admin = User.create(name: "David", address: "123 Test St", city: "Denver", state: "CO", zip: "80204", email: "123@example.com", password: "password", role: 3)
       allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(admin)
 
-      visit "/admin/merchants/#{dog_shop.id}/orders/#{order1.id}"
+      visit "/admin/merchants/#{dog_shop.id}/merchant_orders/#{order1.id}"
       within("#item-#{item_order_1.item.id}") do
         click_on "Fulfill"
       end
-      expect(current_path).to eq("/admin/merchants/#{dog_shop.id}/orders/#{order1.id}")
+      expect(current_path).to eq("/admin/merchants/#{dog_shop.id}/merchant_orders/#{order1.id}")
```
Comment (DavidTTran) [inline spec/features/admin/merchants_spec.rb:211]: I probably could've left the routes as `/admin/merchants/:merchant_id/orders/` with an as block, but for the time being it wasn't recognizing that we had a nested controller for merchant/items

File: spec/features/admin/merchant_items_spec.rb [hunk 3]
```diff
@@ -90,14 +90,17 @@
...
     within "#item-#{paper.id}" do
-      expect(page).to have_no_content("Activate Item")
+      expect(page).to have_content("Active?: true")
+      expect(page).to have_link("Deactivate Item")
       click_on("Deactivate Item")
+      expect(page).to have_content("Active?: false")
+      expect(page).to have_link("Activate Item")
       click_on("Activate Item")
-      expect(page).to have_content("Deactivate Item")
+      expect(page).to have_content("Active?: true")
```
Comment (DavidTTran) [inline spec/features/admin/merchant_items_spec.rb:103]: What'd you do to fix this?

File: config/routes.rb [hunk 4]
```diff
@@ -1,30 +1,34 @@
...
-  post "/merchants", to: "merchants#create"
-  get "/merchants/:id/edit", to: "merchants#edit"
-  patch "/merchants/:id", to: "merchants#update"
-  delete "/merchants/:id", to: "merchants#destroy"
-
-  get "/items", to: "items#index"
-  get "/items/:id", to: "items#show"
-  get "/items/:id/edit", to: "items#edit"
-  patch "/items/:id", to: "items#update"
+  resources :items, only: [:index, :show, :edit, :update, :destroy] do
```
Comment (colintalex) [inline config/routes.rb:13]: Nice job on this

File: config/routes.rb [hunk 5]
```diff
@@ -54,40 +58,48 @@
...
-    delete '/merchants/:id', to: 'merchants#destroy'
-    get '/merchants/:id/items/add-item', to: 'merchant_items#new'
-    post '/merchants/:id/items', to: 'merchant_items#create'
-    get '/merchants/:merchant_id/items/:id/edit', to: 'merchant_items#edit'
-    patch '/merchants/:merchant_id/items/:id', to: 'merchant_items#update'
-    delete '/merchants/:merchant_id/items/:id', to: 'merchant_items#destroy'
-    put '/merchants/:merchant_id/items/:id', to: 'merchant_items#update_active'
-    get '/merchants/:merchant_id/orders/:id', to: 'merchant_orders#show'
-    patch '/merchants/:merchant_id/orders/:order_id/:item_id', to: 'merchant_orders#update'
+    # delete '/merchants/:id', to: 'merchants#destroy'
```
Comment (DavidTTran) [inline config/routes.rb:90]: Just left in the old routes as comments so others knew which ones the resources affected

File: spec/features/admin/merchants_spec.rb (hunk 2 above)
Comment (colintalex) [inline spec/features/admin/merchants_spec.rb:211]: Route helpers really make it easier to understand, definitely glad we used them

File: spec/features/merchants/statistics_spec.rb [hunk 6]
```diff
@@ -40,8 +40,6 @@
           expect(page).to have_content("Denver")
         end
       end
-
```
Comment (colintalex) [inline spec/features/merchants/statistics_spec.rb:43]: I probably made a bunch of these, thanks for the cleanup

**Review by colintalex — APPROVED** [review]
Looks good, route helpers are nicely implemented and makes it easier to read

File: spec/features/admin/merchant_items_spec.rb (hunk 3 above)
Comment (Maxwell-Baird) [reply spec/features/admin/merchant_items_spec.rb:103]: I moved resources to the bottom on the namespace, and added :update to resources :merchant_items, except

**Pull request merged**